src/PostgREST/Config/JSPath.hs

```diff
@@ -1,51 +1,98 @@
+{-# OPTIONS_GHC -Wno-unused-do-bind #-}
 module PostgREST.Config.JSPath
   ( JSPath
   , JSPathExp(..)
+  , FilterExp(..)
   , dumpJSPath
   , pRoleClaimKey
   ) where
 
 import qualified Text.ParserCombinators.Parsec as P
 
-import Data.Either.Combinators       (mapLeft)
-import Text.ParserCombinators.Parsec ((<?>))
-import Text.Read                     (read)
+import Data.Either.Combinators (mapLeft)
+import Text.Read               (read)
 
 import Protolude
 
 
--- | full jspath, e.g. .property[0].attr.detail
+-- | full jspath, e.g. .property[0].attr.detail[?(@ == "role1")]
 type JSPath = [JSPathExp]
 
--- | jspath expression, e.g. .property, .property[0] or ."property-dash"
+-- | jspath expression
 data JSPathExp
-  = JSPKey Text
-  | JSPIdx Int
+  = JSPKey Text         -- .property or ."property-dash"
+  | JSPIdx Int          -- [0]
+  | JSPFilter FilterExp -- [?(@ == "match")], [?(@ ^== "match-prefix")], etc
```

Comment:
How about a test that also has two conditions like the example here?

Reply:
Or if it doesn't make sense to have 2 conditions of this type.. should we restrict it to just 1?

Reply:
Hmm, I am not sure what you are asking here. We are already testing all possible conditions in `io-tests` added in `fixtures.yaml`.

Reply:
I mean, is it possible to have:

```
jwt-role-claim-key = '[?(@ == "match")], [?(@ ^== "match-prefix")]'
```

If so, does it make sense to support that?

Reply:
Oh, ok. I misread the comment. I suggest just having one value to avoid confusion.
```suggestion
  | JSPFilter FilterExp -- [?(@ == "match")]
```

Additionally, from the `type JSPath = [JSPathExp]`, it looks like we can have multiple `JSPFilter`? Could we make it so it only accepts one? 

Or how do we ensure that currently? At the parser level? If so, and if the above is too complicated, maybe add a comment here mentioning it.

Reply:
Yes, this is ensured at the parser level. It only accepts `JSPFilter` at the end of input. I understand it is not the best design but it would have gotten too complicated otherwise. Sure, I'll add a comment mentioning this.
```haskell
pJSPFilter :: P.Parser JSPathExp
pJSPFilter = do
  P.try $ P.string "[?("
  condition <- pFilterConditionParser
  P.char ')'
  P.char ']'
  P.eof -- this should be the last jspath expression
  (return $ JSPFilter condition) <?> "pJSPFilter: JSPath filter exp"

```